lib/factory_bot/factory.rb

```diff
@@ -43,7 +44,11 @@ def run(build_strategy, overrides, &block)
       evaluation =
         Evaluation.new(evaluator, attribute_assigner, compiled_to_create, observer)
 
-      strategy.result(evaluation).tap(&block)
+      evaluation.notify(:before_all, nil)
+      instance = strategy.result(evaluation).tap(&block)
+      evaluation.notify(:after_all, instance)
```

Comment:
Suggestion: if one goal of these callbacks is to disable and then re-enable certain global side effects (like sending emails or other model callbacks), then it might make sense to put this in an `ensure` block, so that it will trigger even if `strategy.result(...)` raises an exception. Without an `ensure`, then a factory creation failure would make it difficult to trigger a reset of global state.

(I would liken this to how `rspec` guarantees the execution of `after` callbacks even when tests fail.)

WDYT?